lib/SILGen/SILGenExpr.cpp

```diff
@@ -1571,8 +1570,6 @@ RValue RValueEmitter::visitOptionalTryExpr(OptionalTryExpr *E, SGFContext C) {
   catchCleanups.pop();
 
   if (isByAddress) {
-    SGF.emitInjectOptionalNothingInto(E,
-                    optInit->getAddressForInPlaceInitialization(SGF, E), optTL);
```

Comment:
I'm unsure about whether this is needed or not. I believe this portion is building the `catch` portion of the try handling, so it seems like it we probably need something to make sure that the result is set to `nil` in the case that an error was thrown.

But I'm also concerned that `optInit` has already had `finishInitialization()` called from within `emitExprInto()` up above. Does this indicate that I shouldn't be using `emitExprInto` above? Or is it okay to call this line after initialization is finished?

Reply:
You're right to be concerned.  There currently isn't any way to re-use an Initialization like this in different control-flow paths, so you need to pattern the code on what emitOptionalEvaluation does, where it creates a new Initialization (with `useBufferAsTemporary`) to emit the expression into, then cancels its cleanup and finishes the original Initialization.

...if you're feeling adventurous, you could try to add this sort of branching support as a feature to Initialization; I think all of our `Initialization` implementations should allow it.